src/_zkapauthorizer/replicate.py

```diff
@@ -207,9 +207,20 @@ class _ReplicationCapableConnection:
     All of this type's methods are intended to behave the same way as
     ``sqlite3.Connection``\ 's methods except they may also add some
     additional functionality to support replication.
+
+    :ivar _replicating: ``True`` if this connection is currently in
+        replication mode and is recording all executed DML statements,
+        ``False`` otherwise.
     """
 
     _conn: Connection
+    _replicating: bool
+
+    def enable_replication(self) -> None:
+        """
+        Turn on replication support.
+        """
+        self._replicating = True
```

Comment:
I like that idea but that way of factoring the code seemed harder to me because of how the different parts of the system interact (with VoucherStore touching the connection directly but the web code needing to be the thing that turns on replication).  I wouldn't mind changing the factoring in the future to simplify all of these pieces and interactions.  I don't think it's worth doing before the replication system actually works though.